llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -5431,7 +5465,7 @@ BoUpSLP::getReorderingData(const TreeEntry &TE, bool TopToBottom) {
 
     // FIXME: Remove the non-power-of-two check once findReusedOrderedScalars
     // has been auditted for correctness with non-power-of-two vectors.
-    if (!TE.isNonPowOf2Vec())
+    if (!TE.hasNonWholeRegisterElems(*TTI))
```

Comment:
If you've audited this for arbitrary VL, please just remove the check.  (In it's own commit.)

Reply:
Reordering requires some extra testing, reverting it for now